### DISCUSSION TITLE
Change fidelity level of physics?

### DISCUSSION THREAD

**Discussion by ShirelJosef** [JSBSim-Team/jsbsim#928] [Q&A]
The main idea is that I want to learn a controller and then check how robust it will be for changes in dynamics 

1. Is there a way to change the fidelity level of the simulation? make it less or more accurate?
2. What properties in JSBSIM can I toggle on/off to make the simulation less/more real? for example...remove/add wind
3. I want to check how robust is my controller, so maybe another option is to learn controller on F15 and then check it on F16?

Any ideas would be great

**Comment by jonsberndt** [comment]
There are probably a few ways to test it. Need a little more information,
though. Are you talking about an autopilot, stability augmentation, heading
hold ... ?

**Reply by ShirelJosef** [reply]
Hey Jon
I am using reinforcement learning to learn autonomous maneuvers with the intent to later deploy it on an actual aircraft there are algorithms to transfer from sim2real.
But instead I want to do something like sim2sim to validate the algorithms and see how fast I learn on new environment (f15 to f16 environment for example)

I hope that’s enough context

**Comment by jonsberndt** [comment]
That helps a very little bit. 

You should be able to find (through trial and error if nothing else) a trim position at altitude and at a given velocity. From there you could have a script that activates a maneuver, setting a target heading, or altitude, or roll rate, or whatever. 

There’s still a lot that is unclear here, though, from a “mechanics” perspective. What are your target parameters? What are the inputs to the autonomous maneuvers?

**Comment by seanmcleod** [comment]
> then check how robust it will be for changes in dynamics...

You could as you suggest train on one aircraft type and then test on a different aircraft type, or modify some of aerodynamic coefficients, e.g. Cm-alpha, Cm-q etc. between training and testing to see how robust your controller is.
